### PR TITLE
Fix getMore with comment specs

### PR DESCRIPTION
**Pull request by comandeo**
https://jira.mongodb.org/browse/DRIVERS-2243 - Fix OperationFailure on MongoDB <4.4.0
https://jira.mongodb.org/browse/DRIVERS-2244 - Test getMore with comment on all topologies

### CODE REVIEW

File: source/change-streams/tests/unified/change-streams.yml [hunk 1]
```diff
@@ -140,7 +140,50 @@ tests:
   - description: "Test that comment is set on getMore"
     runOnRequirements:
       - minServerVersion: "4.4.0"
-        topologies: [ single, replicaset ]
```
Comment (jmikola) [inline source/change-streams/tests/unified/change-streams.yml:143]: I've noted that `single` was introduced by https://github.com/mongodb/specifications/commit/ad1ee9b764f7d9ff193fe485d31feca536e6a2d7; however, this had no effect as the top-level `runOnRequirements` specified replica sets and sharded clusters only (https://github.com/mongodb/specifications/commit/962fb5f20e9e31b528319fe245147928a3100411#diff-ba3c94557f77e83f167fb27ed81eb223e4eb3e6178238f3d93def75f37adb301R3).

File: source/change-streams/tests/unified/change-streams.yml [hunk 2]
```diff
@@ -140,7 +140,50 @@ tests:
...
+        topologies: [ replicaset ]
```
Comment (jmikola) [inline source/change-streams/tests/unified/change-streams.yml:143]: Should you add a comment explaining why sharded clusters are excluded here (e.g. potentially empty `getMore` responses)?

If we do want to include sharded clusters, I suppose we could utilize `ignoreExtraEvents`, which was added in https://github.com/mongodb/specifications/commit/ff0c705f9cc95a5a9c8c595dbf9384cb41dfcdd2 for [DRIVERS-1713](https://jira.mongodb.org/browse/DRIVERS-1713); however, that would then require drivers to implement those test runner changes just to sync these tests.

The best course of action may be to just leave the comment here and open another DRIVERS ticket to come back to these tests and add `ignoreExtraEvents` to allow sharded clusters. That wouldn't be something you'd need to work on yourself -- reporting the issue and leaving it in the backlog would be sufficient just so we don't lose track of it.

**Review by jmikola — APPROVED** [review]
LGTM with added comment to the change stream test. As a sanity check I'd also ask that @ShaneHarvey POC these in Python before we merge.

**Review by baileympearson — APPROVED** [review]
I also tested these changes in the Node driver, the tests look good ([evergreen run](https://spruce.mongodb.com/version/623e083632f4173240ace899/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC))

**Comment by juliusgeo** [comment]
Tested this in the Python driver--[evergreen run]

**Review by ShaneHarvey — APPROVED** [review]

File: source/change-streams/tests/unified/change-streams.yml (hunk 2 above)
Comment (comandeo) [reply source/change-streams/tests/unified/change-streams.yml:146]: Good point, thank you. Created a ticked and added corresponding comment to tests.

**Pull request merged**